src/dynarmic/backend/x64/block_of_code.cpp

```diff
@@ -52,6 +52,28 @@ constexpr size_t CONSTANT_POOL_SIZE = 2 * 1024 * 1024;
 
 class CustomXbyakAllocator : public Xbyak::Allocator {
 public:
+#ifndef _WIN32
+    static constexpr size_t PAGE_SIZE = 4096;
+
+    // Can't subclass Xbyak::MmapAllocator because it is not a pure interface
+    // and doesn't expose its construtor
+    uint8_t* alloc(size_t size) override {
+        // Waste a page to store the size
+        size += PAGE_SIZE;
+        void* p = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
```

Comment:
```suggestion
        void* p = mmap(nullptr, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
```

Also note some operating systems have `MAP_ANON` but not `MAP_ANONYMOUS`.
Also note that macOS requires `MAP_JIT` for hardened runtime support (you can assume it exists as we do not support pre-Mojave macOS).